pkg/operator/controller/status/controller.go

```diff
@@ -570,6 +610,68 @@ func computeGatewayAPICRDsDegradedCondition(state operatorState) configv1.Cluste
 	return degradedCondition
 }
 
+// computeGatewayAPIInstallDegradedCondition computes the degraded condition for the Gateway API OSSM subscription. It
+// checks for known conflicting subscriptions, as well as already existing subscription(s) to OSSM3, and reports
+// degraded when any of those subscriptions would prevent the installation of an appropriate Istio control plane.
+func computeGatewayAPIInstallDegradedCondition(state operatorState) configv1.ClusterOperatorStatusCondition {
+	degradedCondition := configv1.ClusterOperatorStatusCondition{}
+
+	if len(state.Subscriptions) == 0 {
+		return degradedCondition
+	}
+
+	conflicts := []string{}
+	for _, subscription := range state.Subscriptions {
+		if subscription.Spec.Package == "servicemeshoperator3" {
+			isOurSubscription := false
+			for annotation := range subscription.Annotations {
+				if annotation == operatorcontroller.IngressOperatorOwnedAnnotation {
+					isOurSubscription = true
+					break
+				}
+			}
+			if isOurSubscription {
+				// The subscription that the ingress operator creates naturally does not conflict with itself.
+				continue
+			}
+			if subscription.Status.InstalledCSV == "" {
+				// The subscription hasn't finished its install. We will get another reconcile request once the
+				// installation is complete, so we can ignore this for now.
+				continue
+			}
+			versionDiff, err := compareVersionNums(subscription.Status.InstalledCSV, state.expectedGatewayAPIOperatorVerison)
+			if err != nil {
+				log.Error(err, "failed to compare installed OSSM version to expected", "installed", subscription.Status.InstalledCSV, "expected", state.expectedGatewayAPIOperatorVerison)
+				conflicts = append(conflicts, fmt.Sprintf("Installed version %s may prevent enabling operator-managed Gateway API. Install version %s or remove subscription %s/%s to enable.", subscription.Status.InstalledCSV, state.expectedGatewayAPIOperatorVerison, subscription.Namespace, subscription.Name))
+			}
+			switch {
```

Comment:
No need to log the error *and* update `conflicts`; might as well just put the error message in the `conflicts` message.  Also, you don't want to continue to the `switch` if you got an error:
```suggestion
			versionDiff, err := compareVersionNums(subscription.Status.InstalledCSV, state.expectedGatewayAPIOperatorVerison)
			switch {
			case err != nil:
				conflicts = append(conflicts, fmt.Sprintf("failed to compare installed OSSM version to expected: %w", err))
```